docs/conf.py

```diff
@@ -260,6 +263,61 @@
 # Set canonical URL from the Read the Docs Domain
 html_baseurl = os.environ.get("READTHEDOCS_CANONICAL_URL", "")
 
+
+def _custom_edit_url(
+    github_user,
+    github_repo,
+    github_version,
+    doc_path,
+    file_name,
+    default_edit_page_url_template,
+):
+    """Create custom 'edit' URLs for API modules since they are dynamically generated."""
+    if file_name.startswith("api/astropy.") and file_name.endswith(".rst"):
+        # this is a dynamically generated API page
+        astropy_path = file_name.removeprefix("api/astropy.").removesuffix(".rst")
+        item = operator.attrgetter(astropy_path)(astropy)  # noqa: F405
+        if module := getattr(item, "__module__", None):
+            mod_dir, _, mod_file = module.rpartition(".")
+            new_file_name = mod_file + ".py"
+            try:
+                line_no = inspect.findsource(item)[1]
+            except Exception:
+                # Warn if not just a cosmology instance, or a wcs compiled function.
+                if not (
+                    (
+                        "cosmology.realizations" in file_name
+                        and not isinstance(item, type)
+                    )
+                    or "modeling.tabular.Tabular" in file_name
+                    or "astropy.wcs" in file_name
+                ):
+                    warnings.warn(
+                        f"could not find source for {doc_path=}, {file_name=}"
+                    )
+            else:
+                new_file_name += f"#L{line_no + 1}"
+            doc_path = mod_dir.replace(".", "/") + "/"
+            file_name = new_file_name
+        else:
+            if "cosmology.realizations.available" in file_name:
```

Comment:
This is a slippery slope. Once we start to special-case one module, it is bound to blow up in the future.

Reply:
Also I cannot tell just from the diff what is the expected behavior for backport branch, especially after a bunch of refactoring.

Reply:
I agree this all makes it an even worse hack... But I felt there had to be some way to get warnings when no obvious link was available. Also, it seemed better than 404 or essentially giving up!

I really hope the upstream issue can be fixed, so that this all can be removed...

And, indeed, no idea whether the backport will apply cleanly. Though if it is a problem, maybe we can just wait for 7.0? It will be there soon enough...

Reply:
Would be nice to have an option to conditionally hide the edit but my sphinx-fu isn't good enough.

Reply:
Is it a prettier way to "give up" if we redirect to a static HTML page that says edit is disabled for generated API pages? 😅 

I still think an easily breakable edit feature is no better than 404.

Reply:
well, we got a solution that works now, and presumably there will be a proper fix fairly soon(?)

Reply:
I won't hold my breath. This edit thingy was broken even on old theme I think (in a different way). Are you prepared to keep patching it for the foreseeable future?

Reply:
Well, we can always go and remove the edit-on-github if it becomes too annoying (and we're close...)

Reply:
Okay then. I am still technically the PR author, so @mhvk , feel free to approve and merge, and we will see... Thanks!